File: src/utils/logger.js
/**
 * Professional Logger Utility
 * Provides clean, formatted logging with different log levels
 */

const LogLevel = {
  DEBUG: 0,
  INFO: 1,
  SUCCESS: 2,
  WARN: 3,
  ERROR: 4,
};

// ANSI color codes for terminal output
const Colors = {
  reset: '\x1b[0m',
  gray: '\x1b[90m',
  blue: '\x1b[34m',
  green: '\x1b[32m',
  yellow: '\x1b[33m',
  red: '\x1b[31m',
  cyan: '\x1b[36m',
  white: '\x1b[37m',
};

class Logger {
  constructor(module = 'System') {
    this.module = module;
    this.logLevel = LogLevel.DEBUG;
  }

  static create(module) {
    return new Logger(module);
  }

  setLevel(level) {
    this.logLevel = level;
  }

  formatTimestamp() {
    return new Date().toLocaleTimeString('en-US', { hour12: false });
  }

  formatMessage(level, message, data = null) {
    const timestamp = this.formatTimestamp();
    const prefix = `[${timestamp}] [${this.module}]`;
    
    if (data) {
      return `${prefix} ${level}: ${message} ${JSON.stringify(data)}`;
    }
    return `${prefix} ${level}: ${message}`;
  }

  colorize(color, text) {
    return `${color}${text}${Colors.reset}`;
  }

  debug(message, data = null) {
    if (this.logLevel <= LogLevel.DEBUG) {
      console.log(this.colorize(Colors.gray, this.formatMessage('DEBUG', message, data)));
    }
  }

  info(message, data = null) {
    if (this.logLevel <= LogLevel.INFO) {
      console.log(this.colorize(Colors.blue, this.formatMessage('INFO', message, data)));
    }
  }

  success(message, data = null) {
    if (this.logLevel <= LogLevel.SUCCESS) {
      console.log(this.colorize(Colors.green, this.formatMessage('SUCCESS', message, data)));
    }
  }

  warn(message, data = null) {
    if (this.logLevel <= LogLevel.WARN) {
      console.log(this.colorize(Colors.yellow, this.formatMessage('WARN', message, data)));
    }
  }

  error(message, data = null) {
    if (this.logLevel <= LogLevel.ERROR) {
      console.log(this.colorize(Colors.red, this.formatMessage('ERROR', message, data)));
    }
  }

  command(action, details = '') {
    const timestamp = this.formatTimestamp();
    const message = details 
      ? `[${timestamp}] [${this.module}] ▶ ${action}: ${details}`
      : `[${timestamp}] [${this.module}] ▶ ${action}`;
    console.log(this.colorize(Colors.cyan, message));
  }

  response(status, message) {
    const timestamp = this.formatTimestamp();
    const statusIcon = status === 'success' ? '✓' : status === 'error' ? '✗' : '•';
    const color = status === 'success' ? Colors.green : status === 'error' ? Colors.red : Colors.white;
    console.log(this.colorize(color, `[${timestamp}] [${this.module}] ${statusIcon} ${message}`));
  }
}

module.exports = { Logger, LogLevel };

File: self_dc_python/utils/logger.py
"""
Logging utilities for the Discord self-bot.
Uses Rich for colored console output.
"""

import logging
import sys
from typing import Optional

from rich.console import Console
from rich.logging import RichHandler
from rich.theme import Theme

# Custom theme for logging
CUSTOM_THEME = Theme({
    "logging.level.success": "green",
    "logging.level.command": "cyan",
    "logging.level.debug": "dim cyan",
})

console = Console(theme=CUSTOM_THEME)


class ColoredFormatter(logging.Formatter):
    """Custom formatter with colors for different log levels."""

    COLORS = {
        "DEBUG": "dim cyan",
        "INFO": "blue",
        "WARNING": "yellow",
        "ERROR": "red",
        "SUCCESS": "green",
        "CRITICAL": "bold red",
    }

    def format(self, record: logging.LogRecord) -> str:
        # Add color to level name
        level_color = self.COLORS.get(record.levelname, "white")
        record.levelname_colored = f"[{level_color}]{record.levelname}[/{level_color}]"
        return super().format(record)


def setup_logging(name: str, level: Optional[int] = None) -> logging.Logger:
    """
    Set up a logger with RichHandler.

    Args:
        name: Logger name
        level: Logging level (default: INFO)

    Returns:
        Configured logger instance
    """
    logger = logging.getLogger(name)

    if level is None:
        level = logging.INFO

    logger.setLevel(level)

    # Remove existing handlers
    logger.handlers = []

    # Create RichHandler
    handler = RichHandler(
        console=console,
        show_time=True,
        show_path=False,
        rich_tracebacks=True,
    )
    handler.setLevel(level)

    # Set formatter
    formatter = logging.Formatter(
        fmt="[%(asctime)s] [%(name)s] %(message)s",
        datefmt="%H:%M:%S",
    )
    handler.setFormatter(formatter)

    logger.addHandler(handler)

    return logger


class LoggerMixin:
    """Mixin class that provides logger functionality."""

    def __init__(self, name: str):
        self._logger = setup_logging(name)

    @property
    def logger(self) -> logging.Logger:
        """Get the logger instance."""
        return self._logger

    def debug(self, message: str, **kwargs) -> None:
        """Log debug message."""
        self._logger.debug(message, extra=kwargs)

    def info(self, message: str, **kwargs) -> None:
        """Log info message."""
        self._logger.info(message, extra=kwargs)

    def warning(self, message: str, **kwargs) -> None:
        """Log warning message."""
        self._logger.warning(message, extra=kwargs)

    def error(self, message: str, **kwargs) -> None:
        """Log error message."""
        self._logger.error(message, extra=kwargs)

    def success(self, message: str, **kwargs) -> None:
        """Log success message (custom level)."""
        # Use info level but prefix with [SUCCESS]
        self._logger.info(f"[SUCCESS] {message}", extra=kwargs)


# Convenience function
def get_logger(name: str) -> logging.Logger:
    """Get a logger instance."""
    return setup_logging(name)
